File: Python Exams/Basic Exam Collection/Basics Online Exam - 15 and 16 June 2019/05. Series/05. Series.py
budget = float(input())
serials_numbers = int(input())
discount_dict = {'Thrones' : 0.50, 'Lucifer' : 0.60, 'Protector' : 0.70, 'TotalDrama' : 0.80, 'Area' : 0.90}

for _ in range(serials_numbers):
    serial_name = input()
    serial_price = float(input())

    if serial_name in discount_dict.keys():
       serial_price *= discount_dict[serial_name]
    budget -= serial_price


if budget >= 0:
    print(f'You bought all the series and left with {budget:.2f} lv.')
else:
    print(f'You need {abs(budget):.2f} lv. more to buy the series!')


'''
25
6
Teen Wolf
8
Protector
5
TotalDrama
5
Area
4
Thrones
5
Lucifer
9
'''

File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 15 and 16 June 2019/05. Series/05. Series.js
function series(arg) {
    let budget = arg.shift()
    let serialsNumbers = arg.shift()
    const discount_dict = {'Thrones' : 0.50, 'Lucifer' : 0.60, 'Protector' : 0.70, 'TotalDrama' : 0.80, 'Area' : 0.90}

    for (let i = 0; i < serialsNumbers; i++) {
        let serialName = arg.shift()
        let serialPrice = arg.shift()

        if (discount_dict[serialName]) {
            serialPrice *= discount_dict[serialName]
        }
        budget -= serialPrice
    }

    if (budget >= 0)
        return `You bought all the series and left with ${budget.toFixed(2)} lv.`
    return `You need ${Math.abs(budget).toFixed(2)} lv. more to buy the series!`
}

//console.log(series([25, 6, 'Teen Wolf', 8, 'Protector', 5, 'TotalDrama', 5, 'Area', 4, 'Thrones', 5, 'Lucifer', 9]))